go/version/version.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2018 The cap Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+var (
+	// Version holds the complete version number. Filled in at linking time.
+	Version = "0.0.0+unknown"
+
+	// Revision is filled with the VCS (e.g. git) revision being used to build
+	// the program at linking time.
+	Revision = ""
+)
```

Comment:
what fills this in?

Reply:
make
```
REVISION=$(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)

GO_LDFLAGS=-ldflags '-X $(PROJECT)/go/version.Version=$(VERSION) -X (PROJECT)/go/version.Revision=$(REVISION)'
```

Reply:
git rev-parse HEAD  get's written here via the ld flag..